dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorSpirvVulkan.java

```diff
@@ -108,6 +110,29 @@ public List<Event> visitSpirvCmpXchg(SpirvCmpXchg e) {
         return visitVulkanCmpXchg(cmpXchg);
     }
 
+    @Override
+    public List<Event> visitSpirvRMWExtremum(SpirvRmwExtremum e) {
+        e.addTags(Tag.Spirv.MEM_VISIBLE, Tag.Spirv.MEM_AVAILABLE, Tag.Spirv.MEM_NON_PRIVATE);
+        String mo = moToVulkanTag(Tag.Spirv.getMoTag(e.getTags()));
+        Register resultRegister = e.getResultRegister();
+        Register dummy = e.getFunction().newRegister(resultRegister.getType());
+        Load load = newRMWLoadWithMo(dummy, e.getAddress(), Tag.Vulkan.loadMO(mo));
+        Expression originalValue = ExpressionFactory.getInstance().makeIntegerCast(dummy, TypeFactory.getInstance().getArchType(), true);
```

Comment:
Why do we need all these casts? Can we just use the original expressions?

Reply:
The original expressions work fine here, casts are dropped.